app/controllers/api/v1/api_controller.rb

```diff
@@ -11,6 +12,12 @@ class Api::V1::ApiController < ActionController::Base
     render json: {:alert => exception.message}
   end
 
+  def set_websites_from_header
+    app_key = request.headers['HTTP_APP_KEY']
+    app_id = request.headers['HTTP_APP_ID']
+    @current_site ||= Website.find_by_app_id_and_app_key(app_id, app_key)
+  end
+
   def current_site
     if params[:website_id] && current_member
       @current_site ||= current_member.websites.where("websites.id = ?", params[:website_id]).try(:first)
```

Comment:
hai sa facem aici altfel. Hai sa combinam in metoda asta current_site cu metoda aia set_websites_from_header. Fai aici asa:
```
if params[:website_id] && current_member
   @current_site ||= current_member.websites.where("websites.id = ?", params[:website_id]).try(:first)
else
   @current_site ||= Website.find_by_app_id_and_app_key(request.headers['HTTP_APP_ID'], request.headers['HTTP_APP_KEY'])
end
```

In felu asta nu avem nevoie de before_filter deasupra si nici de metoda aia de am facut-o acum